textpipe/doc.py

```diff
@@ -74,23 +91,28 @@ def detect_language(self, hint_language=None):
                                          bestEffort=True)
         return best_guesses[0][1]
 
-    @property
-    def spacy_doc(self):
+    def spacy_doc(self, model_name=None):
         """
-        Create a spacy doc and load the language module
+        Loads a spacy doc or creates one if necessary
 
         >>> doc = Doc('Test sentence for testing text')
-        >>> type(doc.spacy_doc)
+        >>> type(doc.spacy_doc())
         <class 'spacy.tokens.doc.Doc'>
         """
-        return self.spacy_nlp()
-
-    @functools.lru_cache()
-    def spacy_nlp(self):
-        lang = self.language if self.language != 'un' else self.hint_language
-        # loading models with two letter language codes doesn't work for windows
-        spacy_nlp = spacy.load('{}_core_{}_sm'.format(lang, 'web' if lang == 'en' else 'news'))
-        return spacy_nlp(self.clean)
+        lang = self.hint_language if self.language == 'un' else self.language
+        if lang not in self.allowed_languages:
+            raise TextpipeLanguageNotSupportedException(f'The language {lang} is not passed '
+                                                        f'to the pipeline as allowed_language')
+        if model_name in self._spacy_docs:
+            doc = self._spacy_docs[model_name]
+        else:
+            if model_name not in self.spacy_nlps[lang]:
+                raise TextpipeMissingModelException(f'Custom model {model_name} '
+                                                    f'is missing in the pipeline.')
+            else:
+                nlp = self.spacy_nlps[lang][model_name]
```

Comment:
Could we use `self.spacy_nlps[model_name]` here too? From a `Doc` perspective, having `spacy_nlps` for multiple languages does not really make sense. As a doc has a specific language. The logic for picking a specific custom model for a specific language would move to the `Operator` then, I guess.